src/lib.rs

```diff
@@ -89,6 +97,9 @@ pub struct HomeserverConfig {
     /// The URL to communicate to the server with.
     public_baseurl: String,
 
+    /// The registration shared secret, if provided.
```

Comment:
I feel that this should have a default value, to let us register even if the configuration didn't provide a `registration_shared_secret`.